src/Ulid.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/*
+ You may not change or alter any portion of this comment or credits
+ of supporting developers from this source code or any supporting source code
+ which is considered copyrighted (c) material of the original comment or credit authors.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+namespace Xmf;
+
+/**
+ * Generate ULID
+ *
+ * @category  Xmf\Ulid
+ * @package   Xmf
+ * @author    Michael Beck <mambax7@gmail.com>
+ * @copyright 2023 XOOPS Project (https://xoops.org)
+ * @license   GNU GPL 2 or later (https://www.gnu.org/licenses/gpl-2.0.html)
+ */
+class Ulid
+{
+    const ENCODING_CHARS = '0123456789ABCDEFGHJKMNPQRSTVWXYZ';
+    const ENCODING_LENGTH = 32;
+    /**
+     * Last generated timestamp.
+     * @var int
+     */
+    private static $lastTime = 0;
```

Comment:
This change adding the static needs to be reverted. This should be stateless.

The testing would be generating one ULID, followed by a usleep() then generating another ULID  to compare for proper coalation. Best  guess would be usleep(50000)?